app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -26,11 +26,24 @@ var _ component.Component = &Hijacker{}
 
 type MetricsMutator func(in io.Reader, out io.Writer) error
 
+type QueryParametersAppender func(in *url.URL) string
```

Comment:
It's not an appender when you can clear all of the query parameters, more of a `Modifier`? It also receives the whole `URL` so you can pretty much do anything with the `URL` not only with query parameters.